src/components/dashboard/FaceVerification/hooks/useZoomVerification.js

```diff
@@ -0,0 +1,115 @@
+import { useState, useRef, useCallback, useMemo } from "react";
+import { noop, first } from 'lodash';
+
+import api from '../api';
+import { FaceVerificationProviders } from '../api/typings';
+import ZoomAuthentication from "../../../../lib/zoom/ZoomAuthentication";
+
+const sdk = ZoomAuthentication.ZoomSDK;
+const { ZoomSessionStatus, ZoomSession, ZoomCustomization } = sdk;
+
+const getFaceMapBase64 = async faceMetrics => new Promise(
+  (resolve, reject) => faceMetrics.getFaceMapBase64(faceMap =>
+    faceMap ? resolve(faceMetrics)
+      : reject(new Error("Error generating FaceMap !"))
+  )
+)
+
+const initialSessionState = {
+  isComplete: false,
+  isSuccess: false,
+  lastResult: null
+};
+
+export default ({ onComplete = noop, onError = noop }) => {
+  const sessionRef = useRef(null);
+  const sessionResultRef = useRef(null);
+  const sessionSuccessRef = useRef(false);
+  const resultCallbackRef = useRef(null);
+
+  const [sessionState, setSessionState] = useState(initialSessionState);
+
+  const handleCompletion = useCallback(() => {
+    const isSuccess = sessionSuccessRef.current;
+    const lastResult = sessionResultRef.current;
+
+    setSessionState({ isComplete: true, isSuccess, lastResult });
+    onComplete(isSuccess, lastResult);
+  }, [onComplete, setSessionState]);
```

Comment:
only state / props vars and derived (e.g. calculated, formatted etc) from them should be added to deps.
Refs won't affect anything even they'll be added and they're always the same object instances (only .current property changes). So they can be ommitted in deps (checked by exhaustive-deps eslint rule)

I'm performing non-pure component because Zoom handler logic shouldn't cause renders during verification.
So use refs are free-to-use closures to share data between different handlers without state updates when some status changes.
I change state just once, after verification completes (and Zoom unloads its UI)